### ISSUE TITLE
asap queue can be effectively overflown (using q@2.0)

### ISSUE THREAD

**Issue by xytis** [kriskowal/asap#55]
Minimal example snippet:
```
"use strict";

var q = require("q");

var i;
var _handle = function (value) {
    console.log(value);
    return value;
};
for (i = 0; i < 1026; i++) {
    q(i).then(_handle).then(_handle);
}
```

Result (on my machine, 1026 loop iteration fails. I do not know if this is machine dependent)
```
/opt/socketeer/source/node_modules/q/node_modules/asap/asap.js:40
        this.task.call();
                  ^
TypeError: Cannot call method 'call' of null
    at RawTask.call (/opt/socketeer/source/node_modules/q/node_modules/asap/asap.js:40:19)
    at flush (/opt/socketeer/source/node_modules/q/node_modules/asap/raw.js:50:29)
    at process._tickCallback (node.js:415:13)
```

node version: v0.10.25

**Comment by kriskowal** [comment]
Pull requests welcome. I am eager to enable contributors to join this project.

**Comment by ForbesLindesay** [comment]
I've marked asap@2.0.0, 2.0.1 and 2.0.2 as deprecated with a message of "critical race condition fixed in 2.0.3, update now".  Hopefully that will keep people from using the broken versions too often.
